builder/simplify_test.cc

```diff
@@ -130,14 +130,29 @@ TEST(simplify, basic) {
 
 TEST(simplify, let) {
   // lets that should be removed
-  test_simplify(let::make(0, y, z), z);                      // Dead let
-  test_simplify(let::make(0, y * 2, x), y * 2);              // Single use, substitute
-  test_simplify(let::make(0, y, (x + 1) / x), (y + 1) / y);  // Trivial value, substitute
-  test_simplify(let::make(0, 10, x / x), 1);                 // Trivial value, substitute
+  test_simplify(let::make(0, y, z), z);                                // Dead let
+  test_simplify(let::make(0, y * 2, x), y * 2);                        // Single use, substitute
+  test_simplify(let::make(0, y * w, x), y * w);                        // Single use, substitute
+  test_simplify(let::make(0, y, (x + 1) / x), (y + 1) / y);            // Trivial value, substitute
+  test_simplify(let::make(0, 10, x / x), 1);                           // Trivial value, substitute
+  test_simplify(let::make(0, buffer_max(y, 0), x), buffer_max(y, 0));  // buffer_max, substitute
+  test_simplify(let::make(0, buffer_min(y, 0), x), buffer_min(y, 0));  // buffer_min, substitute
+  test_simplify(
+    loop::make(x.sym(), loop_mode::serial, bounds(0, 3), 1, check::make(let::make(0, y, z))),
+    loop::make(x.sym(), loop_mode::serial, bounds(0, 3), 1, check::make(z)));  // Trivial value, substitute
```

Comment:
These new tests still aren't testing what you want to test. The let needs to be defined outside the loop, but used inside the loop, like:
```
let_stmt::make(0, y, z, loop::make(x.sym(), loop_mode::serial, bounds(0, 3), 1, check::make(y))
```

I noticed another problem with these tests: `0` is the same as `x.sym()`. We should use `x.sym()` instead of `0`, and then that makes it obvious that we need to use a different variable for the let (or the loop).